azext_edge/edge/providers/support/base.py

```diff
@@ -504,11 +450,54 @@ def _capture_pod_container_logs(
                         "data": log,
                         "zinfo": (
                             f"{pod_namespace}/{resource_api.moniker}"
-                            f"/pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
+                            f"/{child_group}pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
                         ),
                     }
                 )
             except ApiException as e:
                 logger.debug(e.body)
 
     return processed
+
+
+def _process_kubernetes_resources(
+    resources: object,
+    resource_api: EdgeResourceApi,
+    child_group: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+    kind: Optional[str] = None,
+) -> List[dict]:
+    processed = []
+
+    if not prefix_names:
+        prefix_names = []
+
+    logger.info(f"Detected {len(resources.items)} {kind}s.")
+    for resource in resources.items:
+        r = resource
+        r.api_version = resources.api_version
+        r.kind = kind
+        resource_metadata = r.metadata
+        resource_namespace = resource_metadata.namespace
+        resource_name = resource_metadata.name
+
+        if prefix_names:
+            matched_prefix = [resource_name.startswith(prefix) for prefix in prefix_names]
+            if not any(matched_prefix):
+                continue
+
+        if len(kind) > 12:
```

Comment:
how did you come up with 12 ?

Reply:
pretty sure this is for PersistentVolumeClaim - use if kind == "PersistentVolumeClaim" instead

Reply:
I set 12 because I don't want stateful set to be short termed, but as you mentioned before @vilit1 this short term shouldn't be targe with only PersistentVolumeClaim right? we will probably have long resource in the future to make it short termed, so I set the 12 as threshold.

Reply:
if I use if kind == "PersistentVolumeClaim" then in the future it would also still only work for PersistentVolumeClaim